java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -632,9 +633,24 @@ public CompletableFuture<GlideString[]> mget(@NonNull GlideString[] keys) {
     }
 
     @Override
-    public CompletableFuture<String> mset(@NonNull Map<String, String> keyValueMap) {
-        String[] args = convertMapToKeyValueStringArray(keyValueMap);
-        return commandManager.submitNewCommand(MSet, args, this::handleStringResponse);
+    public <ArgType> CompletableFuture<String> mset(@NonNull Map<ArgType, ArgType> keyValueMap) {
```

Comment:
Better to duplicate code - as I understand, all APIs with `String` would be removed later

Reply:
Since the functions signature differ only in template input Java doesn't allow such overload. I.E this kind of overloading isn't permitted:
someType foo(Map<String,String> ...);
someType foo(Map<GlideString,GlideString> ...);

That's why i had to resort to templating.

Reply:
Yes, Java's "type erasure" thingi